eng/containers/ci.yml

```diff
@@ -64,6 +80,7 @@ jobs:
 
         - task: Docker@2
           displayName: Push ${{ config.name }}:$(imageTag)
+          condition: ne(variables['Build.Reason'], 'PullRequest')
```

Comment:
@scbedd relevant to you, let me know what you think. I'm thinking of adding docker build as a validation step for all containers on PR here.